sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -92,13 +92,16 @@ public void record(double value) {
       View view,
       InstrumentDescriptor instrument,
       Consumer<ObservableLongMeasurement> metricUpdater) {
-    final MetricDescriptor metricDescriptor = MetricDescriptor.create(view, instrument);
+    MetricDescriptor metricDescriptor = MetricDescriptor.create(view, instrument);
     Aggregator<T> aggregator =
         view.getAggregation().createAggregator(instrument, ExemplarFilter.neverSample());
-    final AsyncAccumulator<T> measurementAccumulator = new AsyncAccumulator<>(instrument);
-    final AttributesProcessor attributesProcessor = view.getAttributesProcessor();
+    AsyncAccumulator<T> measurementAccumulator = new AsyncAccumulator<>(instrument, aggregator);
+    if (Aggregator.drop() == aggregator) {
+      return empty();
+    }
```

Comment:
Not sure why this optimization made it to `doubleAsynchronousAccumulator` but not `longAsynchronousAccumulator`.

Got rid of all these the `final` because they reduce readability and don't do anything. If compile time checks are smart enough to know if a `final` variable has been mutated, then the compiler can also be smart enough to apply final optimizations to any variable that is effectively final, but doesn't have the `final` keyword.